src/tools/CMakeLists.txt

```diff
@@ -1,9 +1,18 @@
 
 if (DEFINED ENV{MPAS_TOOL_DIR})
   message(STATUS "*** Using MPAS tools from $ENV{MPAS_TOOL_DIR} ***")
-  file(COPY $ENV{MPAS_TOOL_DIR}/input_gen/namelist_gen DESTINATION .)
-  file(COPY $ENV{MPAS_TOOL_DIR}/input_gen/streams_gen  DESTINATION .)
-  file(COPY $ENV{MPAS_TOOL_DIR}/registry/parse         DESTINATION .)
+  add_custom_target(namelist_gen)
+  add_custom_command(
+    TARGET namelist_gen PRE_BUILD
+    COMMAND ${CMAKE_COMMAND} -E copy $ENV{MPAS_TOOL_DIR}/input_gen/namelist_gen ${CMAKE_CURRENT_BINARY_DIR}/namelist_gen)
+  add_custom_target(streams_gen)
+  add_custom_command(
+    TARGET streams_gen PRE_BUILD
+    COMMAND ${CMAKE_COMMAND} -E copy $ENV{MPAS_TOOL_DIR}/input_gen/streams_gen ${CMAKE_CURRENT_BINARY_DIR}/streams_gen)
+  add_custom_target(parse)
+  add_custom_command(
+    TARGET parse PRE_BUILD
+    COMMAND ${CMAKE_COMMAND} -E copy $ENV{MPAS_TOOL_DIR}/input_gen/parse ${CMAKE_CURRENT_BINARY_DIR}/parse)
 else()
   message(STATUS "*** Building MPAS tools from source ***")
```

Comment:
This change is needed by `./case.build --separate-builds` (aggregated`./case.build` [without separate builds] works fine).

It's not clear why `src/tools/CMakeLists.txt` is not getting picked up while building `ice`: messages are getting logged into e3sm.bldlog, instead of ice.bldlog. But the calls to cmake's `add_custom_target` generate explicit dependencies. Okay.